addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/handler/SamsungTvHandler.java

```diff
@@ -287,14 +172,16 @@ public void reportError(ThingStatusDetail statusDetail, String message, Throwabl
      * devices related to same Samsung TV and create handler for those.
      */
     private void checkAndCreateServices() {
-        logger.debug("Check and create missing UPnP services");
-
-        for (Device device : upnpService.getRegistry().getDevices()) {
-            createService((RemoteDevice) device);
-        }
+        try {
+            logger.debug("Check and create missing UPnP services");
 
-        if (upnpService != null) {
-            upnpService.getRegistry().addListener(this);
+            for (Device device : upnpService.getRegistry().getDevices()) {
+                createService((RemoteDevice) device);
+            }
+        } catch (RuntimeException e) {
+            logger.warn(
+                    "Catching all exceptions because otherwise the checkAndCreateServices thread would silently fail",
```

Comment:
This can be a comment in the code, but should not be the message of the warning.